### ISSUE TITLE
Keyboard not working in search

### ISSUE THREAD

**Issue by jisotalo** [webosbrew/youtube-webos#59]
Hi!

When using external keyboard it isn't working for Youtube search. For all other apps it works and numbers, arrows etc. work also for this. But not alphabetics.

This seems to be problem for original Youtube as well and it's quite old issue.

Do you think it could be fixed with some javascript under the hood? 

Same problem if opening the https://www.youtube.com/tv directly in TVs web browser.

See
* https://support.google.com/youtube/thread/44205654/external-keyboard-unsupported-in-lg-tv-youtube-app?hl=en
* https://support.google.com/youtube/thread/16893541/smart-tv-app-keyboard-functionality?hl=en
* etc..

Awesome app by the way!

**Comment by Informatic** [comment]
Just did a quick look - an issue with YouTube Adfree at least is that Netcast compatiblity layer (which we need to use to rewrite user-agent header...) does not pass keyboard events to the webapp at all. :/

Someone could take a look if it would be feasible to persuade "normal" webappmgr to change its user-agent and use "the official" webapp mode. This would also allow us to fix a bunch of other minor issues we had.

**Comment by Ozzuneoj** [comment]
I would love to see a fix for this. I just rooted my C1 and thought I'd finally start using YouTube on the TV rather than always using the PC, but the lack of keyboard support makes it really annoying to use. I had never really bothered using the official YouTube WebOS app, so it may or may not still do the same thing.

If there is a way to enable physical keyboard support as Informatic said, it would be amazing!

**Comment by Hary309** [comment]
Yeah, it would be nice to have this, the app is very annoying to use.

**Comment by Hary309** [comment]
I have removed the `"trustLevel": "netcast"` from **appinfo.json** and the keyboard started to work. 
Not sure what are the consequences of that because for some reason it was set to it.

**Comment by fire332** [comment]
@Hary309 That breaks the user agent string replacement. What that meant in the past was that the magic remote buttons don't work as expected. Not sure about now though.

**Comment by Hary309** [comment]
I was using it for the past 3 weeks on LG C2 WebOS 24 9.2.1-2605 and I see no issues.

**Comment by fire332** [comment]
@Hary309 can you also remove the user agent replacement line in `appinfo.json` and test to see if that changes anything?

```diff
  "vendorExtension": {
-   "userAgent": "$browserName$/$browserVersion$ ($platformName$-$platformVersion$), _TV_O18/$firmwareVersion$ (LG, $modelName$, $networkMode$)",
    "allowCrossDomain": true
  },
```

**Comment by jisotalo** [comment]
@Hary309 

I tested this now. Edited the `appinfo.json` using ssh and removed that row. The keyboard is still not working in Youtube search / I can't write with it to find something. Only way is to use arrow buttons and enter.

Did you got it working normally?

**Comment by jisotalo** [comment]
OK - it actually seems to work! Perhaps rebooting the app yesterday wasn't enough. Today the keyboard seems to work! I can write text in youtube search normally.

Could this be impelemented in the app as a standard somehow?

Anyways, what I did was edit the `/media/developer/apps/usr/palm/applications/youtube.leanback.v4/appinfo.json` using ssh and removed the `"trustLevel": "netcast"` from the JSON.

**Comment by Hary309** [comment]
@fire332 it doesn't seems to change anything since this field is only used with netcast.
